cpp/src/arrow/type.cc

```diff
@@ -153,8 +153,9 @@ std::string ListType::ToString() const {
 
 MapType::MapType(const std::shared_ptr<DataType>& key_type,
                  const std::shared_ptr<DataType>& item_type, bool keys_sorted)
-    : ListType(struct_({std::make_shared<Field>("key", key_type, false),
-                        std::make_shared<Field>("item", item_type)})),
+    : ListType(std::make_shared<Field>("$data$",
```

Comment:
This is the name that Java has hardcoded as a final variable in the `ListVector`.  It can be changed though, but both sides have to agree or the schema equality test fails if it is different.

Reply:
How about "entries"?

Reply:
Yeah, that sounds much better to me